pulp_smash/tests/rpm/api_v2/test_rsync_distributor.py

```diff
@@ -269,3 +270,94 @@ def test_all(self):
+
+    Do the following:
+
+    * Repeatedly attempt to create an RPM repository with an importer and pair
+      of distributors. Each time, pass an invalid option, or omit a required
+      option.
+    * Create an RPM repository with an importer and pair of distributors. Pass
+      valid configuration options. This demonstrates that creation failures are
+      do to Pulp's validation logic, not some other factor.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a value for the rsync distrib's ``remote`` config section.
+
+        Using the same config for each of the test methods allows the test
+        methods to behave more similarly.
+        """
+        cls.cfg = config.get_config()
+        ssh_user = utils.uuid4()[:12]
+        cls.remote = {
+            'host': 'example.com',
+            'root': '/home/' + ssh_user,
+            'ssh_identity_file': '/' + utils.uuid4(),
+            'ssh_user': ssh_user,
+        }
+        cls._remote = cls.remote.copy()
+
+    def tearDown(self):
+        """Verify that the ``remote`` config section hasn't changed."""
+        # We could also create a setUp() that executes `self.remote =
+        # self.remote.copy()`, thus shadowing the class var. However, that
+        # would merely mask a mis-behaving test method. This assertion lets us
+        # discover it instead.
+        self.assertEqual(self.remote, self._remote)
+
+    def test_success(self):
+        """Successfully create an RPM repo with importers and distributors."""
+        self.make_repo(self.cfg, self.remote)
+
+    def test_required_options(self):
+        """Omit each of the required RPM rsync distributor config options."""
+        for key in self.remote:
+            remote = self.remote.copy()
+            remote.pop(key)
+            with self.subTest(remote=remote):
+                with self.assertRaises(HTTPError):
+                    self.make_repo(self.cfg, remote)
+
+    def test_predistributor_id(self):
+        """Pass a bogus ID as the ``predistributor_id`` config option."""
+        if selectors.bug_is_untestable(2191, self.cfg.version):
+            raise self.skipTest('https://pulp.plan.io/issues/2191')
+        api_client = api.Client(self.cfg, api.json_handler)
+        body = gen_repo()
+        body['importer_config']['feed'] = RPM_FEED_URL
+        body['distributors'] = [gen_distributor()]
+        body['distributors'].append({
+            'distributor_id': utils.uuid4(),
+            'distributor_type_id': 'rpm_rsync_distributor',
+            'distributor_config': {
+                'predistributor_id': utils.uuid4(),
+                'remote': self.remote,
+            }
+        })
+        try:
+            with self.assertRaises(HTTPError, msg=body):
+                repo_href = api_client.post(REPOSITORY_PATH, body)['_href']
+        except AssertionError as err:
+            self.addCleanup(api_client.delete, repo_href)
+            raise err
```

Comment:
This try/except block really does work. You can verify by dropping the `selectors` block at the top of this method.